File: programmsers/그리디/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()  # people 리스트 정렬
    i = 0
    j = len(people) - 1  # 리스트의 길이 -1을 j로 설정
    while i <= j:  # i가 j보다 작을때에만 계속 while문을 돈다.
        if people[i] + people[j] <= limit:  # limit를 아직 넘지 않았을 때
            i += 1
        j -= 1
        answer += 1  # answer는 count 부분

    return answer


people = [70, 50, 80, 50]
limit = 100

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        // 그리디 
        int answer = 0;
        // 최대 2명까지고, 무게 제한이 있는 거다 보니까 투포인터로 사용하는게 맞다는 군
        int left = 0;
        int right = people.length - 1;
        int cnt = 0;
        Arrays.sort(people);
        while(left <= right) {
            if(people[left] + people[right] > limit) {
                cnt++;
                right--;
            } else {
                cnt++;
                right--;
                left++;
            }
        }
        answer = cnt;
        
        return answer;
    }
}